File: backend/src/main/java/com/urmyfood/backend/application/dto/CommentResponse.java
package com.urmyfood.backend.application.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.OffsetDateTime;
import java.util.UUID;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class CommentResponse {

    @JsonProperty("comment_id")
    private UUID commentId;

    @JsonProperty("author_name")
    private String authorName;

    @JsonProperty("author_avatar_url")
    private String authorAvatarUrl;

    private String content;

    @JsonProperty("created_at")
    private OffsetDateTime createdAt;

    @JsonProperty("parent_id")
    private UUID parentId;
}

File: frontend/app-user/src/main/java/com/urmyfood/user/data/model/CommentResponse.kt
package com.urmyfood.user.data.model

import com.google.gson.annotations.SerializedName
import com.urmyfood.user.domain.model.Comment
import java.time.OffsetDateTime
import java.time.format.DateTimeFormatter
import java.util.Locale

data class CommentResponse(
    @SerializedName("comment_id") val commentId: String,
    @SerializedName("author_name") val authorName: String,
    @SerializedName("author_avatar_url") val authorAvatarUrl: String?,
    @SerializedName("content") val content: String,
    @SerializedName("created_at") val createdAt: String
)

private val commentDisplayFormatter = DateTimeFormatter.ofPattern("HH:mm dd/MM/yyyy", Locale.forLanguageTag("vi-VN"))

fun CommentResponse.toDomain() = Comment(
    commentId = commentId,
    authorName = authorName,
    authorAvatarUrl = authorAvatarUrl,
    content = content,
    createdAt = formatCommentTime(createdAt)
)

private fun formatCommentTime(value: String): String =
    runCatching {
        OffsetDateTime.parse(value).format(commentDisplayFormatter)
    }.getOrElse { value }
